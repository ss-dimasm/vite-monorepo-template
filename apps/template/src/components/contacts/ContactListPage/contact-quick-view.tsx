import { FC } from 'react'
import { KeyValueList, StatusIndicator } from '@reapit/elements'
import { ContactModel } from '@reapit/foundations-ts-definitions'
import { combineName } from '../../../utils/combine-name'
import { combineAddress } from '../../../utils/combine-address'

export interface ContactQuickViewProps {
  contact: ContactModel
}

export const ContactQuickView: FC<ContactQuickViewProps> = ({ contact }) => {
  const { forename, title, active, surname, homePhone, workPhone, mobilePhone, email, primaryAddress } = contact
  return (
    <KeyValueList
      items={[
        {
          key: 'Name',
          value: combineName(title, forename, surname),
          iconName: 'contact',
        },
        {
          key: 'Email',
          value: email || '-',
          iconName: 'email',
        },
        {
          key: 'Active Status',
          value: (
            <>
              <StatusIndicator intent={active ? 'success' : 'danger'} /> {active ? 'Active' : 'Inactive'}{' '}
            </>
          ),
          iconName: 'info',
        },
        {
          key: 'Mobile',
          value: mobilePhone || '-',
          iconName: 'phone',
        },
        {
          key: 'Home',
          value: homePhone || '-',
          iconName: 'phone',
        },
        {
          key: 'Work',
          value: workPhone || '-',
          iconName: 'phone',
        },
        {
          key: 'Primary Address',
          value: combineAddress(primaryAddress),
          iconName: 'property',
        },
      ]}
    />
  )
}
