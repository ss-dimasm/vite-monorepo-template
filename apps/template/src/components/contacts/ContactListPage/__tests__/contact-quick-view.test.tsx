import { mockContactModel } from '../../../../scripts/stubs'
import { render } from '../../../../scripts/tests'
import { ContactQuickView } from '../contact-quick-view'

describe('ContactQuickView', () => {
  it('should match a snapshot with data', () => {
    expect(render(<ContactQuickView contact={mockContactModel} />).asFragment()).toMatchSnapshot()
  })
})
