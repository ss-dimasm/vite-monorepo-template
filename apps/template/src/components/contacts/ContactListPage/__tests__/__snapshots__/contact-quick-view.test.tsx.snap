// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactQuickView > should match a snapshot with data 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-93 el-snack-holder"
  />
  <div
    role="list"
  >
    <div
      class="mocked-styled-174 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-173 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon-new el-icon"
        >
          <svg
            fill="none"
            height="1em"
            role="img"
            title="Icon image with name contact"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 15c-4.42 0-8 2.24-8 5h16c0-2.76-3.58-5-8-5m0-11a3.49 3.49 0 0 0-3.5 3.5c.002.754.069 1.507.2 2.25A3.48 3.48 0 0 0 12 13a3.48 3.48 0 0 0 3.3-3.25c.131-.743.198-1.496.2-2.25A3.49 3.49 0 0 0 12 4"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Name
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          Mr Foo Bar
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-174 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-173 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon-new el-icon"
        >
          <svg
            fill="none"
            height="1em"
            role="img"
            title="Icon image with name email"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 4H4a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V6a2 2 0 0 0-2-2m-1.41 2L12 12.59 5.41 6zM4 18V7.41L10.59 14a2 2 0 0 0 2.82 0L20 7.41V18z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Email
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          foo@bar.com
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-174 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-173 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon-new el-icon"
        >
          <svg
            fill="none"
            height="1em"
            role="img"
            title="Icon image with name info"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10A10 10 0 0 0 12 2m1.143 15.714h-2.286v-6h2.286zm-1.143-8a1.714 1.714 0 1 1 0-3.428 1.714 1.714 0 0 1 0 3.428"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Active Status
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          <span
            class="el-intent-success el-status-indicator"
          />
           Active 
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-174 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-173 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon-new el-icon"
        >
          <svg
            fill="none"
            height="1em"
            role="img"
            title="Icon image with name phone"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 16.81a3.3 3.3 0 0 0-.3-.81l-6-1a3.5 3.5 0 0 0-.85 1.78 10.74 10.74 0 0 1-6.65-6.63A3.5 3.5 0 0 0 9 9.29l-1-6a3.2 3.2 0 0 0-.83-.24A3 3 0 0 0 6.6 3 3.6 3.6 0 0 0 3 6.6C3 14.553 9.447 21 17.4 21a3.6 3.6 0 0 0 3.6-3.6 4 4 0 0 0 0-.59"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Mobile
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          242423424
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-174 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-173 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon-new el-icon"
        >
          <svg
            fill="none"
            height="1em"
            role="img"
            title="Icon image with name phone"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 16.81a3.3 3.3 0 0 0-.3-.81l-6-1a3.5 3.5 0 0 0-.85 1.78 10.74 10.74 0 0 1-6.65-6.63A3.5 3.5 0 0 0 9 9.29l-1-6a3.2 3.2 0 0 0-.83-.24A3 3 0 0 0 6.6 3 3.6 3.6 0 0 0 3 6.6C3 14.553 9.447 21 17.4 21a3.6 3.6 0 0 0 3.6-3.6 4 4 0 0 0 0-.59"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Home
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          234234234
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-174 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-173 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon-new el-icon"
        >
          <svg
            fill="none"
            height="1em"
            role="img"
            title="Icon image with name phone"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 16.81a3.3 3.3 0 0 0-.3-.81l-6-1a3.5 3.5 0 0 0-.85 1.78 10.74 10.74 0 0 1-6.65-6.63A3.5 3.5 0 0 0 9 9.29l-1-6a3.2 3.2 0 0 0-.83-.24A3 3 0 0 0 6.6 3 3.6 3.6 0 0 0 3 6.6C3 14.553 9.447 21 17.4 21a3.6 3.6 0 0 0 3.6-3.6 4 4 0 0 0 0-.59"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Work
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          2342342344
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-174 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-173 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon-new el-icon"
        >
          <svg
            fill="none"
            height="1em"
            role="img"
            title="Icon image with name property"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m22.923 9.165-10.5-7a1 1 0 0 0-1.1 0l-10.45 7a1 1 0 0 0 1.1 1.66l.9-.63v11.3a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1v-11.3l.95.63a.94.94 0 0 0 .55.17 1 1 0 0 0 .55-1.83m-4.05 11.33h-14V8.865l7-4.67 7 4.67z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Primary Address
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          Bar Foo Foo Bar Foo Bar Foo
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
